addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/PiiScanRule.java

```diff
@@ -156,20 +157,21 @@ private static boolean isDecimal(String containingString) {
         return containingString.contains(".");
     }
 
-    private void raiseAlert(HttpMessage msg, String evidence, String cardType, BinRecord binRec) {
+    private AlertBuilder createAlert(String evidence, String cardType, BinRecord binRec) {
         String other = Constant.messages.getString(MESSAGE_PREFIX + "extrainfo", cardType);
         if (binRec != null) {
             other = other + '\n' + getBinRecString(binRec);
         }
-        newAlert()
+        return newAlert()
                 .setRisk(Alert.RISK_HIGH)
                 .setConfidence(binRec != null ? Alert.CONFIDENCE_HIGH : Alert.CONFIDENCE_MEDIUM)
                 .setDescription(Constant.messages.getString(MESSAGE_PREFIX + "desc"))
                 .setOtherInfo(other)
                 .setEvidence(evidence)
+                .setSolution(Constant.messages.getString(MESSAGE_PREFIX + "soln"))
+                .setAlertRef(binRec != null ? getPluginId() + "-2" : getPluginId() + "-1")
```

Comment:
I'm not sure about this, it's the same alert, that (more) info about the card is provided does not make it other alert.

Reply:
It's also different confidence.

Reply:
I'm willing to be convinced/forced one way or the other, I don't feel strongly about it.

Reply:
Different confidence does not make it other alert either.

Reply:
@psiinon @ricekot feelings on this? I'm fine either way, just need some sort of consensus.

I guess the one benefit of **not** treating them differently is that we only need a simple example and the commonlib change could be reverted.

Reply:
https://github.com/zaproxy/zaproxy/pull/6278/files#diff-60d722bd9ee62edce83e87ebce5f5b93e6491a52476404a3d3dbd2ad30362850R945-R968

Reply:
Okay I see what you’re getting at. So I’ll drop it back to a single ref.

What about keeping the example verbose? (Ex: keeping the commonlib change)

Reply:
Sounds good.